### PR TITLE
Catch throwables in mview updating

### PR DESCRIPTION
**Pull request by QuentinFarizon**
2.2 : https://github.com/magento/magento2/pull/23124
2.3 : https://github.com/magento/magento2/pull/23125

### Description
Certain type of runtime exception weren't caught during mview updating, which caused some mview to be stuck in "processing" mode.
Refer to https://github.com/magento/magento2/issues/23054 for more detail about the issue.

This modification is very similar to the one made here : https://github.com/magento/magento2/commit/5927a75169aa5eb7a978847d22dbde5dbe3cc5a2

### Fixed Issues
https://github.com/magento/magento2/issues/23054

### Manual testing scenarios
make an mview updating crash with a Throwable (not an Exception)
process ends, mview is still in 'processing' in database
it is never picked up again

### Questions or comments
This does not prevent against all types of crashes (like process out of memory or server stop abruptly, so it is only a step in making indexing crons more resilient

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/Mview/View.php [hunk 1]
```diff
@@ -277,40 +277,51 @@ public function update()
...
+                        0,
+                        $exception
+                    );
+                }
                 throw $exception;
             }
         }
     }
 
+    private function executeAction($action, $currentVersionId, $lastVersionId) {
```
Comment (orlangur) [inline lib/internal/Magento/Framework/Mview/View.php]: Typehints, please, and declare strict types should also work, I believe.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Mview/View.php (hunk 1 above)
Comment (QuentinFarizon) [reply lib/internal/Magento/Framework/Mview/View.php]: @orlangur Done

Comment (orlangur) [reply lib/internal/Magento/Framework/Mview/View.php]: @QuentinFarizonAfrimarket please add `declare(strict types)` on top and if builds will be green - squash into a single commit. If not, don't add such line and just squash.

Comment (QuentinFarizon) [reply lib/internal/Magento/Framework/Mview/View.php]: Done, tests pass with strict type

**Review by orlangur — CHANGES_REQUESTED** [review]
@QuentinFarizonAfrimarket nice! Please just make a single commit out of it.

**Comment by hostep** [comment]
I would honestly keep it as 2 separate commits, they do 2 different things (one bug fix, another is cleanup), so it makes sense to keep them separate in my opinion.

**Comment by orlangur** [comment]
@hostep ah, ok, I didn't notice bugfix + refactoring are separated, thought that was in first commit and then some adjustments.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5210](https://jira.corp.magento.com/browse/ENGCOM-5210) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@QuentinFarizonAfrimarket thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by QuentinFarizon** [comment]
Yes it was my intention to separate behaviour and refactoring :)

**Comment by engcom-Foxtrot** [comment]
QA passed.

**Pull request merged**